server/src/main/java/org/candlepin/liquibase/PoolTypeUpgradeTask.java

```diff
@@ -0,0 +1,196 @@
+        // Store the connection's auto commit setting, so we may temporarily clobber it.
+        boolean autocommit = this.connection.getAutoCommit();
+        this.connection.setAutoCommit(false);
+
+        this.executeUpdate(
+            "UPDATE cp_pool SET type = 'ENTITLEMENT_DERIVED' WHERE cp_pool.id IN (" +
+            "  SELECT P.id " +
+            "  FROM cp_pool P " +
+            "    INNER JOIN cp_pool_attribute PA ON (P.id = PA.pool_id AND PA.name = 'pool_derived') " +
+            "  WHERE " +
+            "    PA.value = 'true' " +
+            "    AND (P.sourceentitlement_id IS NOT NULL AND p.sourceentitlement_id != '')" +
+            ")"
+        );
+
+        this.executeUpdate(
+            "UPDATE cp_pool SET type = 'STACK_DERIVED' WHERE cp_pool.id IN (" +
+            "  SELECT P.id " +
+            "  FROM cp_pool P " +
+            "    INNER JOIN cp_pool_attribute PA ON (P.id = PA.pool_id AND PA.name = 'pool_derived') " +
+            "    INNER JOIN cp_pool_source_stack SS ON p.id = SS.derivedpool_id " +
+            "  WHERE " +
+            "    PA.value = 'true' " +
+            "    AND (P.sourceentitlement_id IS NULL OR p.sourceentitlement_id = '')" +
+            ")"
+        );
+
+        this.executeUpdate(
+            "UPDATE cp_pool SET type = 'BONUS' WHERE cp_pool.id IN (" +
+            "  SELECT P.id " +
+            "  FROM cp_pool P " +
+            "    INNER JOIN cp_pool_attribute PA ON (P.id = PA.pool_id AND PA.name = 'pool_derived') " +
+            "    LEFT JOIN cp_pool_source_stack SS ON p.id = SS.derivedpool_id " +
+            "  WHERE" +
+            "    PA.value = 'true' " +
+            "    AND (P.sourceentitlement_id IS NULL OR p.sourceentitlement_id = '')" +
+            "    AND SS.id IS NULL" +
+            ")"
+        );
+
+        this.executeUpdate(
+            "UPDATE cp_pool SET type = 'UNMAPPED_GUEST' WHERE cp_pool.id IN (" +
+            "  SELECT P.id " +
+            "  FROM cp_pool P " +
+            "    LEFT JOIN cp_pool_attribute PA1 ON (P.id = PA1.pool_id AND PA1.name = 'pool_derived') " +
+            "    INNER JOIN cp_pool_attribute PA2 ON (P.id = PA2.pool_id AND PA2.name = 'unmapped_guest') " +
+            "  WHERE PA1.id IS NULL AND PA2.value = 'true'" +
+            ")"
+        );
+
+        this.executeUpdate(
+            "UPDATE cp_pool SET type = 'NORMAL' WHERE cp_pool.id IN (" +
+            "  SELECT P.id " +
+            "  FROM cp_pool P " +
+            "    LEFT JOIN cp_pool_attribute PA1 ON (P.id = PA1.pool_id AND PA1.name = 'pool_derived') " +
+            "    LEFT JOIN cp_pool_attribute PA2 ON (P.id = PA2.pool_id AND PA2.name = 'unmapped_guest') " +
+            "  WHERE PA1.id IS NULL AND PA2.id IS NULL" +
+            ")"
+        );
+
+        // Commit & restore original autocommit state
+        this.connection.commit();
+        this.connection.setAutoCommit(autocommit);
+    }
+
+}
```

Comment:
This looks like just straight SQL statements, could this just be a regular liquibase update without the custom class?